### ISSUE TITLE
Error processing image

### ISSUE THREAD

**Issue by soana** [damongolding/immich-kiosk#127]
**Describe the bug**
Per the error message it looks like app has an issue communicating with the immich server
<img width="365" alt="Screenshot 2024-10-10 at 6 09 03 PM" src="https://github.com/user-attachments/assets/06ba2572-63e0-4b0e-a558-0fb73e0d6362">

**To Reproduce**
After an update that was applied today the issue started happening right away

**Your Kiosk version**
Version 0.11.2

**Your Kiosk installation**
- Docker

**Your Kiosk ENV or config file**
<?xml version="1.0"?>
<Container version="2">
  <Name>immich-kiosk</Name>
  <Repository>damongolding/immich-kiosk:latest</Repository>
  <Registry>https://hub.docker.com/r/damongolding/immich-kiosk/</Registry>
  <Network>bridge</Network>
  <MyIP/>
  <Shell>sh</Shell>
  <Privileged>false</Privileged>
  <Support>https://hub.docker.com/r/damongolding/immich-kiosk/</Support>
  <Project>https://github.com/damongolding/immich-kiosk</Project>
  <Overview>A kiosk for your Immich images&#xD;
&#xD;
Converted By Community Applications   Always verify this template (and values)  against the support page for the container&#xD;
&#xD;
https://hub.docker.com/r/damongolding/immich-kiosk/&#xD;
&#xD;
repository official:&#xD;
damongolding/immich-kiosk:latest&#xD;
&#xD;
repository development&#xD;
damongolding/immich-kiosk-development:latest</Overview>
  <Category>MediaApp:Photos</Category>
  <WebUI>http://[IP]:[PORT:3000]</WebUI>
  <TemplateURL/>
  <Icon>https://github.com/damongolding/immich-kiosk/blob/main/assets/logo.png?raw=true</Icon>
  <ExtraParams/>
  <PostArgs/>
  <CPUset/>
  <DateInstalled>1728598021</DateInstalled>
  <DonateText/>
  <DonateLink/>
  <Requires/>
  <Config Name="Container Port 1" Target="3000" Default="3000" Mode="tcp" Description="" Type="Port" Display="always" Required="false" Mask="false">3000</Config>
  <Config Name="Community Applications Conversion" Target="Community_Applications_Conversion" Default="true" Mode="{3}" Description="" Type="Variable" Display="always" Required="false" Mask="false">true</Config>
  <Config Name="TimeZone" Target="TZ" Default="" Mode="" Description="Time zone" Type="Variable" Display="always" Required="false" Mask="false">America/Detroit</Config>
  <Config Name="Immich API Key" Target="KIOSK_IMMICH_API_KEY" Default="" Mode="" Description="The API for your Immich server." Type="Variable" Display="always" Required="false" Mask="true">***********************************</Config>
  <Config Name="Immich URL" Target="KIOSK_IMMICH_URL" Default="" Mode="" Description="The URL of your Immich server, e.g. http://192.168.1.123:2283." Type="Variable" Display="always" Required="true" Mask="false">http://***.***.***.***:8088</Config>
  <Config Name="Disable_UI" Target="KIOSK_DISABLE_UI" Default="FALSE" Mode="" Description="A shortcut to set show_time, show_date, show_image_time and image_date_format to false." Type="Variable" Display="always" Required="false" Mask="false">FALSE</Config>
  <Config Name="KIOSK_SHOW_DATE" Target="KIOSK_SHOW_DATE" Default="TRUE" Mode="" Description="Display the date." Type="Variable" Display="always" Required="false" Mask="false">TRUE</Config>
  <Config Name="KIOSK_DATE_FORMAT" Target="KIOSK_DATE_FORMAT" Default="" Mode="" Description="The format of the date. default is day/month/year. Any GO date string is valid." Type="Variable" Display="always" Required="false" Mask="false">DD MMM YYYY</Config>
  <Config Name="KIOSK_SHOW_TIME" Target="KIOSK_SHOW_TIME" Default="TRUE" Mode="" Description="Display clock." Type="Variable" Display="always" Required="false" Mask="false">TRUE</Config>
  <Config Name="KIOSK_TIME_FORMAT" Target="KIOSK_TIME_FORMAT" Default="24" Mode="" Description="Display clock time in either 12 hour or 24 hour format. Can either be 12 or 24." Type="Variable" Display="always" Required="false" Mask="false">24</Config>
  <Config Name="KIOSK_REFRESH" Target="KIOSK_REFRESH" Default="" Mode="" Description="The amount in seconds a image will be displayed for." Type="Variable" Display="always" Required="false" Mask="false">30</Config>
  <Config Name="KIOSK_ALBUM" Target="KIOSK_ALBUM" Default="********-****-****-****-***********" Mode="" Description="The Immich ID of a specific album you want to display." Type="Variable" Display="always" Required="false" Mask="false">********-****-****-****-***********</Config>
  <Config Name="KIOSK_PERSON" Target="KIOSK_PERSON" Default="" Mode="" Description="The ID(s) of a specific person or people you want to display. Having the album set will overwrite this. See FAQ: How do I set multiple people? to see how to impliment this." Type="Variable" Display="always" Required="false" Mask="false">********-****-****-****-***********, ********-****-****-****-***********, ********-****-****-****-***********</Config>
  <Config Name="KIOSK_IMAGE_FIT" Target="KIOSK_IMAGE_FIT" Default="CONTAIN" Mode="" Description="cover |contain | none &#13;&#10;How your image will fit on the screen. Default is contain." Type="Variable" Display="always" Required="false" Mask="false">CONTAIN</Config>
  <Config Name="KIOSK_BACKGROUND_BLUR" Target="KIOSK_BACKGROUND_BLUR" Default="TRUE" Mode="" Description="Display a blurred version of the image as a background." Type="Variable" Display="always" Required="false" Mask="false">TRUE</Config>
  <Config Name="KIOSK_TRANSITION" Target="KIOSK_TRANSITION" Default="" Mode="" Description="Which transition to use when changing images. &#9;&#13;&#10;none | fade | cross-fade" Type="Variable" Display="always" Required="false" Mask="false">FADE</Config>
  <Config Name="KIOSK_SHOW_PROGRESS" Target="KIOSK_SHOW_PROGRESS" Default="" Mode="" Description="Display a progress bar for when image will refresh." Type="Variable" Display="always" Required="false" Mask="false">TRUE</Config>
  <Config Name="KIOSK_SHOW_IMAGE_TIME" Target="KIOSK_SHOW_IMAGE_TIME" Default="" Mode="" Description="Display image time from METADATA (if available)." Type="Variable" Display="always" Required="false" Mask="false">TRUE</Config>
  <Config Name="KIOSK_IMAGE_TIME_FORMAT" Target="KIOSK_IMAGE_TIME_FORMAT" Default="" Mode="" Description="Display image time in either 12 hour or 24 hour format. Can either be 12 or 24." Type="Variable" Display="always" Required="false" Mask="false">24</Config>
  <Config Name="KIOSK_SHOW_IMAGE_DATE" Target="KIOSK_SHOW_IMAGE_DATE" Default="" Mode="" Description="Display the image date from METADATA (if available)." Type="Variable" Display="always" Required="false" Mask="false">TRUE</Config>
  <Config Name="KIOSK_IMAGE_DATE_FORMAT" Target="KIOSK_IMAGE_DATE_FORMAT" Default="" Mode="" Description="The format of the image date. default is day/month/year. Any GO date string is valid." Type="Variable" Display="always" Required="false" Mask="false">DD MMM YYYY</Config>
  <Config Name="KIOSK_SHOW_IMAGE_LOCATION" Target="KIOSK_SHOW_IMAGE_LOCATION" Default="true|false" Mode="" Description="KIOSK_SHOW_IMAGE_LOCATION" Type="Variable" Display="always" Required="false" Mask="false">true</Config>
  <Config Name="KIOSK_SLEEP_START" Target="KIOSK_SLEEP_START" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">23</Config>
  <Config Name="KIOSK_SLEEP_END" Target="KIOSK_SLEEP_END" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">630</Config>
  <Config Name="KIOSK_FONT_SIZE" Target="KIOSK_FONT_SIZE" Default="100" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">200</Config>
  <Config Name="KIOSK_THEME" Target="KIOSK_THEME" Default="fade | solid" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">fade </Config>
  <Config Name="KIOSK_LAYOUT" Target="KIOSK_LAYOUT" Default="single | splitview" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false"> splitview</Config>
</Container>

**Any parms passed to the URL used to access Kiosk**
- no extra parameters

**Raspberry Pi:**
 - Raspberry Pi OS Release 2024-07-04 runing in Wayland kiosk mode (planing to switch to X11)
 - Chrome

**MacOS**
 - Sonoma
 - Firefox 131.0

**Additional context**
I can confirm that Immich server is OK and running and that the albums and persons are present on the Immich side

For some reason the docker xml file does not up properly, below are some screenshots:
 
<img width="867" alt="Screenshot 2024-10-10 at 6 42 40 PM" src="https://github.com/user-attachments/assets/95059844-ee09-412b-8acf-8d233260b577">
<img width="783" alt="Screenshot 2024-10-10 at 6 42 59 PM" src="https://github.com/user-attachments/assets/ec9d8d5d-f3f9-46bf-b872-11deaa7de4c5">
<img width="893" alt="Screenshot 2024-10-10 at 6 42 51 PM" src="https://github.com/user-attachments/assets/75416dd4-ca03-4963-9bc7-17c229ea30f1">
<img width="775" alt="Screenshot 2024-10-10 at 6 42 25 PM" src="https://github.com/user-attachments/assets/02b2b303-8a45-4610-81b2-150b68a4e572">
<img width="845" alt="Screenshot 2024-10-10 at 6 42 13 PM" src="https://github.com/user-attachments/assets/926d78ed-505d-470a-b533-a29f3d8b60b3">
<img width="515" alt="Screenshot 2024-10-10 at 6 42 01 PM" src="https://github.com/user-attachments/assets/a241d1c1-418e-490e-ac3c-8c1a4406dac3">

**Comment by soana** [comment]
I can confirm that: damongolding/immich-kiosk-development:latest is working just fine.

**Comment by damongolding** [comment]
It looks like person and album are empty in the unraid GUI but have values in the output above 🤔. Are you adding either albums or people?


I should also note that I have no idea who created and manages this unraid template.

**Comment by damongolding** [comment]
But in essence the error stems from Kiosk trying to retrieve a person's asset count from Immich to use for assets weighting. 

In this case as status of 400 has been sent back from Immich, which means the request was malformed. 

Is the person ID correct? It can change if you make certain changes to a person I.e. merge two people. 

I get the same error if I set person to a ID that doesn't exist.

**Comment by soana** [comment]
Both album and person fields are there I just wiped them off for the screenshots, this is my personal template I created before an template was available in the unraid apps store.

After several trails I figured out that the issue comes from the KIOSK_PERSON variable. Up until recently it used to work with a space after the comma separating multiple persons but lately that did not work anymore giving the error.
